### DISCUSSION TITLE
Intelligent templates of paragraphs or clauses

### DISCUSSION THREAD

**Discussion by ykburda** [TriliumNext/Trilium#6053] [General]
Hi
First of all thanks any members of Trilliumnext for all kind of efforts. I have an idea for paragraph templates. Imagine we have a contract document. It has some specific clauses eg : 1.1 safety requirments 1.2 legal requirements so on. I would like to have these clauses as templates. And If i’m creating a new X contract , i will add eg: 1.1 , 1.3 , 1.4 and it will be automatically embed to the document with latest version of template. Here embedding means copy and linking a relation to clause template. It is not just showing clause template. This is the first requirement.

Later imagine i have changed clauses template to a new version just changing with wording. Somehow i need seeing relations of this clause and i will re-create a new version of X contract. Old contracts of couse will stay as the version of the original clause template till i create a new contract version.

I would like to discuss this amazing idea. I believe this will be a major upgrade of trillium notes.

**Comment by eliandoran** [comment]
Greetings, @ykburda .

Sounds like an interesting idea but it also seems quite specific to a particular use case. I would say that for the particular use case described, it would make more sense to have a dedicated application handling the contract creation rather than a personal knowledge based tool.

Nevertheless, I think at least the first part with paragraph templates would be interesting. The editor that we are using is called CKEditor and it has support for templates (see [demo](https://ckeditor.com/docs/ckeditor5/latest/features/template.html)), but they are a paid feature so that means we cannot benefit from it yet. I guess those paragraphs could be implemented using this template function if we ever get to implement it, depending on the number of templates (I don't expect it to work fine if there are hundreds of different templates).

Regarding linking the template instance (so the text that remains the same) to the template, I'm not convinced on how useful it would be. You could simply prepend an ID of the template (or even an actual link to it) and have it removed in the final document (either by hiding it via CSS if you are exporting to HTML then PDF or find & replace if you are using Markdown).

**Comment by ykburda** [comment]
Hi Elian,
I gave a sample for contracts. But it’s also any kind of documents of legal document or human resources document or any procedures.

 If the program has some targets  it should cover many functions. If we say this is only a personel editor then it’s all right. Notepad, wordpad has many more. But this program may have additional functions and users will select to use it. Here a new chance. Assume you have many template paragraphs and they are assigned to docs. Just need to follow new version required or not. If you create a new version of inherited document you will get latest.

**Comment by meichthys** [comment]
@ykburda You can also investigate using the Trilium "Include Note" feature. This lets you embed (and display) a note inside of another note.
If you print the note, it will also print the 'included' note. 

![image](https://github.com/user-attachments/assets/df8de1a7-90fd-47a8-94f1-d50d077a8ddf)

**Reply by eliandoran** [reply]
I would suppose that this does not fit @ykburda 's use case since the Include note makes the included text not directly editable inside the new note, and changing the template would change the documents using it as well.

**Reply by ykburda** [reply]
I need to inherit any version  of template to a note. I can change template to a new version. I need to see all related notes and i may revise (or may not) to new version of linked note. Kind of inherit template with version.
